R/aggregate_profiles.R

```diff
@@ -352,6 +353,10 @@ aggregated_profiles_conditional <- function(all_profiles, groups = NULL, span =
       # for categorical variables we will calculate weighted average
       # but weights are 0-1, 1 if it's the same level and 0 otherwise
       split_profile$`_w_` <- split_profile$`_orginal_` == split_profile$`_x_`
+
+      # for factors, keep proper order
+      # as in https://github.com/ModelOriented/ingredients/issues/82
+      split_profile$`_x_` <- factor(split_profile$`_x_`, levels = unique(split_profile$`_x_`))
```

Comment:
this way, the reordering will be don in each step of the `lapply` loop. In each step `_x_` will have different levels (these are different variables). I would keep the releveling after the `lappy`, it should be faster (done once) and less error-prone (we do not assume how factors with different levels will be merged)